include/exadg/time_integration/time_int_abm_base.h

```diff
@@ -0,0 +1,266 @@
+                           is_test_),
+      pde_operator(pde_operator_in),
+      // order of predictor can be chosen one below order of corrector
+      ab(order_ - 1, start_with_low_order_),
+      am(order_, start_with_low_order_),
+      vec_evaluated_operators(order_ - 1)
+  {
+    AssertThrow(order_ >= 1,
+                dealii::ExcMessage("Oder of ABM time integrator has to be at least 1."));
+  }
+
+  void
+  print_iterations() const
+  {
+    // explicit time integration -> one iteration
+    print_list_of_iterations(pcout, {"Adams-Bashforth-Moulton"}, {1});
+  }
+
+  virtual bool
+  print_solver_info() const = 0;
+
+  void
+  ale_update()
+  {
+    AssertThrow(false, dealii::ExcMessage("not yet implemented"));
+  }
+
+  VectorType const &
+  get_solution() const
+  {
+    return solution;
+  }
+
+private:
+  void
+  update_time_integrator_constants() final
+  {
+    ab.update(time_step_number, adaptive_time_stepping, time_steps);
+    am.update(time_step_number, adaptive_time_stepping, time_steps);
+  }
+
+  void
+  allocate_vectors() final
+  {
+    pde_operator->initialize_dof_vector(solution);
+    pde_operator->initialize_dof_vector(prediction);
+
+    pde_operator->initialize_dof_vector(evaluated_operators_np);
+    for(auto & evaluated_operator : vec_evaluated_operators)
+      pde_operator->initialize_dof_vector(evaluated_operator);
+  }
+
+  void
+  initialize_current_solution() final
+  {
+    pde_operator->prescribe_initial_conditions(solution, get_time());
+  }
+
+  void
+  initialize_former_multistep_dof_vectors() final
+  {
+    if(start_with_low_order)
+    {
+      if(vec_evaluated_operators.size() > 0)
+        evaluate_operator(vec_evaluated_operators[0], solution, get_time());
+
+      // first prediction (the time loop starts with the correction)
+      pde_operator->prescribe_initial_conditions(solution, get_time());
+      predrict_solution(prediction, solution, vec_evaluated_operators);
+    }
+    else // !start_with_low_order
+    {
+      std::vector<VectorType> vec_temp_u_grid(vec_evaluated_operators.size());
+      // fill evaluated operators
+      VectorType temp_sol;
+      pde_operator->initialize_dof_vector(temp_sol);
+      for(unsigned int i = 0; i < vec_evaluated_operators.size(); ++i)
+      {
+        pde_operator->prescribe_initial_conditions(temp_sol, get_previous_time(i));
+        evaluate_operator(vec_evaluated_operators[i], temp_sol, get_previous_time(i));
+      }
+
+      // first prediction (we start with correction in the time loop)
+      pde_operator->prescribe_initial_conditions(solution, get_time());
+      predrict_solution(prediction, solution, vec_evaluated_operators);
+    }
+  }
+
+  void
+  setup_derived() final
+  {
+  }
+
+  virtual double
+  calculate_time_step_size() = 0;
+
+  virtual double
+  recalculate_time_step_size() const = 0;
+
+  void
+  do_timestep_solve() final
+  {
+    dealii::Timer timer;
+    timer.restart();
+
+    // predict operator for next time step
```

Comment:
This comment seems somewhat imprecise. What I would like more is e.g. "evaluate operator given the predicted solution"?

Reply:
Done in [7a46047](https://github.com/exadg/exadg/pull/585/commits/7a460471668fbef3a38a2f3721bd820f0b14be2a)